service/sources/ProdusController.cpp

```diff
@@ -0,0 +1,61 @@
+//
+// Created by Tudor Coman on 03.04.2022.
+//
+
+#include "../headers/ProdusController.h"
+
+ProdusController::ProdusController(const std::string &path) : Controller(path) { }
+
+void ProdusController::handle_get(const http_request &req) {
+    std::map<std::string, std::string> http_get_vars = uri::split_query(req.request_uri().query());
+    auto output = Utilities::getJsonArray<Produs>(&produsRepository, http_get_vars);
+    req.reply(status_codes::OK, output);
+}
+
+void ProdusController::handle_post(const http_request &req) {
+    pplx::task<web::json::value> jsonTask = req.extract_json();
+    web::json::value json = jsonTask.get();
+    Produs p;
+    p.fromJson(json);
+
+    try {
+        produsRepository.getById(p.getId());
+        // update operation
+        produsRepository.opUpdate(p.getId(), p);
+        req.reply(status_codes::OK, "Produsul a fost creat");
+    } catch(const std::runtime_error& e) {
+        if(std::string(e.what()) == "ID does not exist") {
```

Comment:
nu e ok, mai bine prinzi excepția de not found cu un catch separat și nu faci comparații pe mesaje

comparațiile pe mesaje le faci în situații rare, când nu prea ai de ales (de exemplu când interacționezi cu biblioteci externe)